ompi/info/info.c

```diff
@@ -116,76 +116,85 @@ int ompi_mpiinfo_init(void)
     return OMPI_SUCCESS;
 }
 
-/*
- * Fill in the MPI_INFO_ENV if using MPI3 initialization
+/**
+ * Fill in an info object with ENV info.  Used for setting
+ * MPI_INFO_ENV and by invocation of MPI_Info_create_env.
  */
-int ompi_mpiinfo_init_mpi3(void)
+
+int ompi_mpiinfo_init_env(int argc, char **argv, ompi_info_t *info)
 {
-    char *cptr, **tmp;
+    char *cptr, **tmp = NULL;
 
     /* fill the env info object */
 
     /* command for this app_context */
-    if (NULL != ompi_process_info.command) {
+    if (NULL != argv) {
+        tmp = argv;
+    } else if (NULL != ompi_process_info.command) {
         tmp = opal_argv_split(ompi_process_info.command, ' ');
-        opal_info_set(&ompi_mpi_info_env.info.super, "command", tmp[0]);
+    }
+
+    if (NULL != tmp) {
+        opal_info_set(&info->super, "command", tmp[0]);
```

Comment:
@hppritcha Can't `tmp[0]` be NULL? Could it be an empty string? In any of these case, does it make sense to set the `"command"` key?

Reply:
is your test invoking MPI_Info_create_env prior to calling mpi_session_init or mpi_init/init_thread?  I'm having problems reproducing this.  I agree we should check if tmp[0] is NULL.

Reply:
> is your test invoking MPI_Info_create_env prior to calling mpi_session_init or mpi_init/init_thread?

No, `MPI_Init_thread` is called first. I can try to reproduce locally, but not today.

Reply:
Anyway, check the MPI 4.0 pp. 513. `MPI_Info_create_env` may be called BEFORE `MPI_Init[_thread]` or `MPI_Session_init`.